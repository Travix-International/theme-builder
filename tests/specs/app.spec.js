const app = require('../../src/app.js');
const yaml = require('js-yaml');

describe('App', () => {
  it('should process yaml file with proper processor with prefix', () => {
    const doc = {
      do: {
        some: {
          string: 'aaa',
        }
      }
    };

    yaml.safeLoad = () => doc;
    const result = app('some.yaml', 'scss', { prefix: 'tx' });
    expect(result).toMatchSnapshot();
  });

  it('should process yaml file with processor', () => {
    const doc = {
      do: {
        some: {
          string: 'aaa',
        }
      }
    };

    yaml.safeLoad = () => doc;
    const result = app('some.yaml', 'js');
    expect(result).toMatchSnapshot();
  });

  it('should return null in case of missing processor', () => {
    yaml.safeLoad = () => {};
    const result = app('some.yaml', 'no-processor');
    expect(result).toBeNull();
  });
});
