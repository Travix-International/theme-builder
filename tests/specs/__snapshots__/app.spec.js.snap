exports[`App should process yaml file with processor 1`] = `
Object {
  "do": Object {
    "some": Object {
      "string": "aaa",
    },
  },
}
`;

exports[`App should process yaml file with proper processor with prefix 1`] = `
Array [
  "$tx-do-some-string: aaa;",
]
`;
